labels-config-service-impl/src/main/java/org/hypertrace/label/config/service/LabelsConfigServiceImpl.java

```diff
@@ -83,7 +83,8 @@ public void createLabel(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       CreateLabel createLabel = request.getLabel();
-      if (systemLabelsKeyLabelMap.containsKey(createLabel.getKey())) {
+      if (systemLabelsKeyLabelMap.containsKey(createLabel.getKey())
```

Comment:
You need synchronisation across these multiple calls. Refer to https://github.com/hypertrace/config-service/blob/1a5f85c34f62ce96b901ef11f16b36c1a23f44b6/config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java#L45 for syncronization across critical section. Even this doesn't work on multi-pod deployment. However we can start with this.